ukpn/scripts/resample_data.py

```diff
@@ -0,0 +1,94 @@
+"""Function to resample the irreggular time series data into regular"""
+import random
+from typing import Optional, Tuple
+
+import matplotlib.pyplot as plt
+import pandas as pd
+from dateutil.parser import parse
+
+
+def resample_dataframe(
+    path_to_file: str, intervals: Optional[str] = "5Min", resample: Optional[bool] = False
+) -> pd.DataFrame:
+    """This function resamples a time series into regular intervals
+
+    Args:
+        path_to_file: Enter the absolute path to the csv file
+        intervals: Enter the intervals that you want this data to have
+        resample: Option to resample the dataframe
+
+    """
+    # Reading the csv data from the path
+    df = pd.read_csv(path_to_file, header=None, names=["date_time", "bad_data"], sep=",")
+
+    # # Convert data values from str into int
+    df["bad_data"] = pd.to_numeric(df["bad_data"], errors="coerce")
+
+    # Interpolate with padding
+    df["bad_data"] = df["bad_data"].interpolate(method="pad", limit=2)
+
+    # Converting into datetime format
+    df["date_time"] = df["date_time"].apply(parse)
+    
+    # Reset index
+    df = df.reset_index(drop=True)
+
+    # If resample is True
+    if resample:
+
+        # Rounding to nearest 5 minutes
+        df["date_time"] = df["date_time"].dt.round("5Min")
```

Comment:
So for this, and the interpolate above, I wouldn't interpolate before resampling for this. And I wouldn't round these date times to 5 minutes. What would be better would be to create a pandas series of 5 minutely intervals covering the time period

```suggestion
        time_series_to_interpolate = pd.date_range(start="2021-10-15 00:00", end="2022-05-31 00:00", freq="5m")
```

And then use something like https://docs.scipy.org/doc/scipy/reference/generated/scipy.interpolate.interp1d.html#scipy.interpolate.interp1d to interpolate from the raw data points to the 5 minutely spaced values. Then there shouldn't be any duplicate values to drop. Another way would be like in this post: https://stackoverflow.com/questions/18955250/interpolating-one-time-series-onto-another-in-pandas to keep it all in pandas, and interpolate between the date times.

Reply:
Sure, I'll follow this! Thanks!

Reply:
Please check the PR, I have written a interpolation function and tested it.